simulator-core/src/main/java/com/evbox/everon/ocpp/simulator/station/StationState.java

```diff
@@ -153,13 +152,53 @@ public Evse findEvse(int evseId) {
                     .findAny();
     }
 
+    /**
+     * Try to find an EVSE with no ongoing transaction.
+     *
+     * @return optional with instance of {@link Evse} or Optional.empty()
+     */
+    public Optional<Evse> tryFindAvailableEvse() {
+        return evses.values()
+                    .stream()
+                    .filter(evse -> !evse.hasOngoingTransaction())
+                    .findAny();
+    }
+
     public Optional<Connector> tryFindConnector(int evseId, int connectorId) {
         return tryFindEvse(evseId)
                 .flatMap(evse -> evse.getConnectors().stream()
                         .filter(connector -> connector.getId().equals(connectorId))
                         .findAny());
     }
 
+    /**
+     * Saves the future for a scheduled connection timeout.
+     *
+     * @param evseId id of the evse
+     * @param future future of the scheduled timeout
+     */
+    public void saveConnectionTimeOutFuture(int evseId, ScheduledFuture future) {
+        connectionTimeOutFutures.put(evseId, future);
+    }
+
+    /**
+     * Stops the running future that will trigger the connection
+     * timeout for transactions started remotely.
+     *
+     * @param evseId id of the evse
+     * @return true if the evse had a future running
+     */
+    public boolean stopConnectionTimeOut(int evseId) {
```

Comment:
How about checking if cable is plugged in a Future and call stopTransaction only if cable is not plugged? Would it be possible to avoid having this method then?